### PR TITLE
manual_tests: change macro loader to TEST

### LINKED ISSUE #7426: [Tests] `scenarios/manual_tests/scenario-test.cfg` unconsciously loads code into core
and triggers false-positives in log.

```ini
previously defined at campaigns/Under_the_Burning_Suns/utils/abilities.cfg:595
    included from test/scenarios/manual_tests/scenario-test.cfg:10
    included from _main.cfg:54
20230228 12:59:49 warning preprocessor: Redefining macro ABILITY_DISENGAGE without explicit #undef at ~add-ons/War_of_Legends/macros/abilities-new.cfg:179
    included from ~add-ons/War_of_Legends/_main.cfg:25
previously defined at campaigns/Under_the_Burning_Suns/utils/abilities.cfg:621
    included from test/scenarios/manual_tests/scenario-test.cfg:10
    included from _main.cfg:54
20230228 12:59:49 warning preprocessor: Redefining macro ABILITY_DISTRACT without explicit #undef at ~add-ons/War_of_Legends/macros/abilities.cfg:164
    included from ~add-ons/War_of_Legends/_main.cfg:25
previously defined at campaigns/Under_the_Burning_Suns/utils/abilities.cfg:643
    included from test/scenarios/manual_tests/scenario-test.cfg:10
    included from _main.cfg:54
20230228 12:59:49 warning preprocessor: Redefining macro WEAPON_SPECIAL_DAZE without explicit #undef at ~add-ons/War_of_Legends/macros/nightmare-utils.cfg:96
    included from ~add-ons/War_of_Legends/_main.cfg:25
previously defined at campaigns/Under_the_Burning_Suns/utils/abilities.cfg:631
    included from test/scenarios/manual_tests/scenario-test.cfg:10
    included from _main.cfg:54
20230228 12:59:49 warning preprocessor: Redefining macro WEAPON_SPECIAL_DAZE_EVENTS without explicit #undef at ~add-ons/War_of_Legends/macros/nightmare-utils.cfg:104
    included from ~add-ons/War_of_Legends/_main.cfg:25
previously defined at campaigns/Under_the_Burning_Suns/utils/abilities.cfg:686
    included from test/scenarios/manual_tests/scenario-test.cfg:10
    included from _main.cfg:54
```

These are all inside SP-campaigns and should NOT be loading them in MP.

### PR DESCRIPTION
**Pull request by knyghtmare**
Closes #7426

### CODE REVIEW

**Comment by Pentarctagon** [comment]
Looks like there's a WML validation issue in the code behind that #ifdef. I assume [resource] isn't valid to be inside [units] since it's a top level tag?

**Comment by knyghtmare** [comment]
> I assume [resource] isn't valid to be inside [units] since it's a top level tag?

`[units]` is a top-level tag. `[resource]` is normally defined inside `#ifdef` of content. Perhaps that is a reason why

**Comment by CelticMinstrel** [comment]
Isn't that block of code completely redundant? Aren't those tests loaded further down in the other TEST block?

**Comment by Pentarctagon** [comment]
No.

**Comment by CelticMinstrel** [comment]
Okay, it would appear you're correct, but in that case… wouldn't it make sense to move this to data/test/_main.cfg?

**Comment by Pentarctagon** [comment]
Probably this is another result of me not understanding why we have two entirely different sets of tests that can only be run manually.

**Comment by CelticMinstrel** [comment]
What are you talking about? There's only one set of tests that can be run manually, isn't there? There's also the AI demo scenarios, which are also defined as test scenarios, but I'd consider them demos rather than tests. (But they can still somewhat function as manual test scenarios as well.)

**Comment by Pentarctagon** [comment]
The `-t` tests vs the manual `-u` tests such as test_max_menu_items.

**Comment by stevecotton** [comment]
All the `-t` tests can be run manually with `-u` too, although victory/defeat isn't how those tests report results, so a dialog saying "You have been defeated" doesn't necessarily mean the test failed. Running them with `-u` helps with debugging why a test is failing.

The difference between them is that
* the `-t` ones can automatically do all the actions and then work out whether the results match the expectations
* some `-u` tests need someone to manually look at them and say whether they passed or failed
* other `-u` tests are really workbenches for developers to do some action and then work out if the result matched their expectations. They need someone to manually do some actions.

**Comment by Pentarctagon** [comment]
> the -t ones can automatically do all the actions and then work out whether the results match the expectations

What? Even the default test scenario (just `wesnoth -t`) doesn't work like that - it's a big scenario where you can manually trigger all sorts of actions.

**Comment by stevecotton** [comment]
Oops, I've got and swapped round `-t` and `-u` in my comment. I'll fix it.

**Comment by Pentarctagon** [comment]
How is `-t` different from `-u --show-gui`?

**Comment by CelticMinstrel** [comment]
> The `-t` tests vs the manual `-u` tests such as test_max_menu_items.

Um, "manual `-u`" tests are not a thing. Any test can technically be run with either `-u` or `-t`, but you're not supposed to run manual tests with `-u`.

> How is `-t` different from `-u --showgui`?

`-t` doesn't return a result code that specifies a pass or fail. `-u` does, even with `--showgui`. The manual test scenarios are not configured to have a result in the form of victory or defeat – they're set up so that you can do things and view the outcome. Therefore, calling them with `-u` would just make the game hang, and calling them with `-u --showgui` also isn't useful because the returned result will be meaningless.

**Comment by Pentarctagon** [comment]
`test_max_menu_items` and `test_relative_dir` were in the same place as the rest of the automated WML unit tests thus some of the confusion, though admittedly that's no longer the case. Would it be correct to say that:
* Everything in `data/test/scenarios/manual_tests` is intended to be run with `-t`
* All other tests in sub-folders under `data/test/scenarios` are primarily meant to be run with `-u` (for the automated WML unit tests)

?

**Comment by stevecotton** [comment]
> `test_max_menu_items` and `test_relative_dir` were in the same place as the rest of the automated WML unit tests thus some of the confusion, though admittedly that's no longer the case.

I'd look at it the other way round - scenario-leaders.cfg, scenario-movethrough.cfg, scenario-story.cfg and scenario-test.cfg should all have been created in data/test, (or moved there when data/test was created, whichever came afterwards).

>     * Everything in `data/test/scenarios/manual_tests` is intended to be run with `-t`
>     * All other tests in sub-folders under `data/test/scenarios` are primarily meant to be run with `-u` (for the automated WML unit tests)

I thought that was the design of your reorg, and I agree with it. I haven't double-checked if it's currently true though, there might be a stray manual test somewhere.

**Comment by CelticMinstrel** [comment]
I do think leaving the AI demo scenarios where they are is fine, though. They are meant to be run with `-t`, but they're also slightly more "user-facing" than all the other test scenarios.

File: data/_main.cfg [hunk 1]
```diff
@@ -49,7 +49,7 @@
 #endif
 [/ais]
 
-#ifdef DEBUG_MODE
+#ifdef TEST
```
Comment (Pentarctagon) [inline data/_main.cfg:52]: The contents of this #ifdef can be added to the already existing one starting on line 63.

Comment (CelticMinstrel) [reply data/_main.cfg:52]: Or moved into data/test/_main.cfg

**Comment by Pentarctagon** [comment]
So for the WML schema error, it looks like the [resource] defined at the end of `Under_the_Burning_Suns/utils/abilities.cfg` needs to be moved to a separate file?

**Comment by CelticMinstrel** [comment]
I guess it could go in _main.cfg?

**Comment by stevecotton** [comment]
I think we should just take Kaleh out of scenario-test.cfg. He was added in 34d48bd251d7d98ee0579dd98380705bceb596d7, which seems to just be a way to test his AMLAs without playing the campaign itself. Why not just test in an UtBS scenario?

I'm against changing UtBS' code for this - if the scenario-test wants Quenoth Elves, then it should have the code to cope with them. If scenario-test doesn't load that resource then it means you can create elves with debug mode, but their abilities won't work.

**Comment by stevecotton** [comment]
> moved

I meant "completely removed" :D

**Comment by CelticMinstrel** [comment]
I'm saying that the test scenario is including the abilities.cfg file inside `[units]` but should be including it at toplevel.

**Comment by CelticMinstrel** [comment]
Doesn't the latest change mean that `-t` ceases to function entirely?

Why not just address the actual issue, that the test scenario includes abilities.cfg inside `[units]`?

**Pull request closed without merging**